control/grpc.py

```diff
@@ -236,42 +260,92 @@ def __init__(self, config: GatewayConfig, gateway_state: GatewayStateHandler, rp
         self.subsys_max_ns = {}
         self.host_info = SubsystemHostAuth()
 
-    def create_host_psk_file(self, subsysnqn : str, hostnqn : str, psk_value : str) -> str:
+    def create_host_key_file(self, key_type : str, subsysnqn : str, hostnqn : str, key_value : str) -> str:
         assert subsysnqn, "Subsystem NQN can't be empty"
         assert hostnqn, "Host NQN can't be empty"
-        assert psk_value, "PSK value can't be empty"
+        assert key_value, "Key value can't be empty"
 
-        psk_dir = f"/tmp/psk/{subsysnqn}"
-        psk_file = f"{psk_dir}/{hostnqn}"
+        key_dir = f"/tmp/{key_type}/{subsysnqn}"
+        key_file = f"{key_dir}/{hostnqn}"
         try:
-            os.makedirs(psk_dir, 0o755, True)
+            os.makedirs(key_dir, 0o755, True)
```

Comment:
Please don't use direct file methods, [Python provides a library](https://docs.python.org/3/library/tempfile.html) to deal with temporary dirs/files securely, including automatic cleanup of files.

Reply:
I don't want to automatically delete the files as they still needed later for the connection. Will use the Python tempfile.